x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/topn/TopNOperator.java

```diff
@@ -273,6 +276,12 @@ public String describe() {
     private final List<SortOrder> sortOrders;
 
     private Row spare;
+    /**
+     * Esp. in case of many fields, a row's values can become quite large; growing from an empty row every time is expensive due
+     * to lots of allocations. Instead, keep track of the last row's actual size and use that to pre-allocate memory for the next row.
+     */
+    private int lastSeenRowValuesSize = 32;
```

Comment:
Maybe this should start at 0 - in case that the sort keys are all we have, we might be unnecessarily allocating data for the values.

Reply:
I'm good starting it at 0, yeah. I could absolutely see a world where we're only sorting, like, a stream of `int`s or `keyword`s.